### ISSUE TITLE
Netlify CMS failing GraphQL query

### ISSUE THREAD

**Issue by thecodelayeriser** [gatsbyjs/gatsby#7508]
Hi guys, 

I'm running a website with Gatsby and Netflify CMS. I'm also using GraphQL to do queries for data I want to show. I'm having a problem that I can not solve - I have read the earlier posted issues with the same problem and while I have tried to follow the solutions proposed I'm not able to solve my problem. 

Basically, when doing a query with 

`{
  allMarkdownRemark(sort: {fields: [frontmatter___date], order: DESC}) {
    edges {
      node {
        id
        frontmatter {
          title
          date(formatString: "DD MMMM YYYY")
          image {
            childImageSharp {
              responsiveSizes(maxWidth: 400) {
                src
                srcSet
                sizes
              }
            }
          }
        }
        excerpt
        fields {
          slug
        }
        html
        excerpt
      }
    }
  }
}`

I'm getting an unexpected result: 

`{
  "errors": [
    {
      "message": "Field \"image\" must not have a selection since type \"String\" has no subfields.",
      "locations": [
        {
          "line": 9,
          "column": 17
        }
      ]
    }
  ]`

My repo is available here: https://github.com/Layerise/Website - is it possible to get some feedback?

In short, my gatsby-node.js looks like this:

```
const path = require('path');
const { createFilePath } = require('gatsby-source-filesystem');

exports.onCreateNode = ({ node, getNode, actions }) => {
  const { createNodeField } = actions;
  if (node.internal.type === 'MarkdownRemark') {
    const slug = createFilePath({
      node,
      getNode,
      basePath: 'posts',
    });
    createNodeField({
      node,
      name: 'slug',
      value: `/resources${slug}`,
    });
  }
};

exports.createPages = ({ graphql, actions }) => {
  const { createPage } = actions;
  return new Promise((resolve, _reject) => {
    graphql(`
      {
        allMarkdownRemark {
          edges {
            node {
              fields {
                slug
              }
            }
          }
        }
      }
    `).then(result => {
      result.data.allMarkdownRemark.edges.forEach(({ node }) => {
        createPage({
          path: node.fields.slug,
          component: path.resolve('./src/components/PostPage.js'),
          context: {
            slug: node.fields.slug,
          },
        });
      });
      resolve();
    });
  });
};
```

My gatsby-config.js looks like this:

```
module.exports = {
  siteMetadata: {
    title: 'Layerise',
    description: 'This is our descriptions text',
    keywords: 'keyword line',
    siteUrl: 'https://layerise.netlify.com',
  },
  plugins: [
    `gatsby-plugin-styled-components`,
    'gatsby-plugin-offline',
    'gatsby-plugin-react-helmet',
    'gatsby-transformer-remark',
    'gatsby-transformer-sharp',
    'gatsby-plugin-sharp',
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `pages`,
        path: `${__dirname}/src/pages/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `posts`,
        path: `${__dirname}/src/posts/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `img`,
        path: `${__dirname}/static/images/`,
      },
    },
    {
      resolve: 'gatsby-plugin-favicon',
      options: {
        logo: './src/images/favicon.png',

        // WebApp Manifest Configuration
        appName: 'Layerise',
        appDescription: null,
        developerName: null,
        developerURL: null,
        dir: 'auto',
        lang: 'en-US',
        background: '#010032',
        theme_color: '#010032',
        display: 'standalone',
        orientation: 'any',
        start_url: '/',
        version: '1.0',

        icons: {
          android: true,
          appleIcon: true,
          appleStartup: true,
          coast: false,
          favicons: true,
          firefox: true,
          opengraph: false,
          twitter: false,
          yandex: false,
          windows: false,
        },
      },
    },
    {
      resolve: `gatsby-plugin-google-analytics`,
      options: {
        trackingId: 'UA-120447155-1',
        // Puts tracking script in the head instead of the body
        head: false,
        // Setting this parameter is optional
        anonymize: true,
        // Setting this parameter is also optional
        respectDNT: true,
        // Avoids sending pageview hits from custom paths
        exclude: ['/preview/**', '/do-not-track/me/too/'],
      },
    },
    {
      resolve: `gatsby-plugin-netlify-cms`,
      options: {
        /**
         * One convention is to place your Netlify CMS customization code in a
         * `src/cms` directory.
         */
        stylesPath: `${__dirname}/src/cms/cms.js`,
      },
    },
  ],
};
```

My config.yml looks like this:

```
backend:
  name: github
  repo: Layerise/Website

media_folder: static/images
public_folder: /images
publish_mode: editorial_workflow

collections:
  - name: post
    label: Posts
    folder: src/posts
    create: true
    slug: '{{year}}-{{slug}}'
    fields:
      - { name: title, label: Title, required: true }
      - { name: date, label: Create At, widget: date }
      - { name: body, label: Post, widget: markdown, required: true }
      - { name: image, label: Thumbnail, widget: image, required: true }
```

I'm sure I'm missing something obvious. But after many attempts some guidance would be awesome.

**Comment by LeKoArts** [comment]
And where do you do this query? I couldn't find it.

**Comment by thecodelayeriser** [comment]
Hi, I'm using this query in GraphQL.

**Comment by thecodelayeriser** [comment]
To help as much as I can I have added a branch with my current ideas: https://github.com/Layerise/Website/tree/Gatsby-plugin-for-cms

Resources.js

```
import React from 'react';
import Layout from '../components/layout';
import styled from 'styled-components';
import { graphql } from 'gatsby';
import PostListing from '../components/PostListing';
import Heading from '../components/Heading';

const MasonryGrid = styled.div`
  column-width: 504px;
  column-gap: 48px;
  margin: 0 auto;
  > li {
    margin-bottom: 48px;
  }
  @media (max-width: 1056px) {
    column-width: 324px;
    column-gap: 24px;
  }
  @media (max-width: 360px) {
    column-gap: 24px;
  }
`;

const PaddingBottom = styled.div`
  padding-bottom: 200px;
`;

const Resources = ({ data }) => (
  <Layout>
    <Heading
      H1="Research results"
      P="Results form our global surveys and findings."
    />
    <PaddingBottom />
    <MasonryGrid>
      {data.allMarkdownRemark.edges.map(({ node }) => (
        <PostListing key={node.id} post={node} />
      ))}
    </MasonryGrid>
  </Layout>
);

export default Resources;

export const query = graphql`
  query SiteMeta {
    site {
      siteMetadata {
        title
        description
      }
    }
    allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
      edges {
        node {
          id
          frontmatter {
            title
            date(formatString: "DD MMMM YYYY")
            image {
              childImageSharp {
                fluid(maxWidth: 630) {
                  ...GatsbyImageSharpSizes
                }
              }
            }
          }
          excerpt
          fields {
            slug
          }
          html
          excerpt
        }
      }
    }
  }
`;
```

Postlisting.js

```
import React from 'react';
import Link from 'gatsby-link';
import styled from 'styled-components';
import P from './P';
import Date from './Date';
import Img from 'gatsby-image';

const Article = styled.li`
  background: #f9f9f9;
  display: inline-block;
  max-width: 504px;
`;

const ArticleContainer = styled.div`
  padding: 60px;
  > h1 {
    margin-top: 60px;
    margin-bottom: 24px;
  }
`;

const H1 = styled.h1`
  letter-spacing: -0.2px;
  font-weight: 700;
  font-size: 36px;
  letter-spacing: -1.5px;
  line-height: 62px;
  text-decoration: none;
  @media (max-width: 700px) {
    font-size: 32px;
    letter-spacing: -0.86px;
    line-height: 38px;
  }
`;

const StyledLink = styled(
  styled(Link)`
    &:hover {
      text-decoration: underline;
    }
  `,
  'active',
)`
  color: #010032;
`;

const ArticleImg = styled.div`
  width: 204px;
  > img {
    width: 100%;
  }
`;

const PostListing = ({ post }) => (
  <Article>
    <ArticleContainer>
      <ArticleImg>
        {/* <img src={post.frontmatter.image} alt="" /> */}
        <Img fluid={post.frontmatter.image.childImageSharp.fluid} alt="" />
      </ArticleImg>
      <H1>
        <StyledLink to={post.fields.slug}>{post.frontmatter.title}</StyledLink>
      </H1>
      <Date>{post.frontmatter.date}</Date>
      <P>{post.excerpt}</P>
    </ArticleContainer>
  </Article>
);

export default PostListing;
```

Gatsby-config.js

```
module.exports = {
  siteMetadata: {
    title: 'Layerise',
    description: 'This is our descriptions text',
    keywords: 'keyword line',
    siteUrl: 'https://layerise.netlify.com',
  },
  plugins: [
    `gatsby-plugin-styled-components`,
    'gatsby-plugin-offline',
    'gatsby-plugin-react-helmet',
    'gatsby-transformer-sharp',
    'gatsby-plugin-sharp',
    {
      resolve: `gatsby-transformer-remark`,
      options: {
        plugins: [
          {
            resolve: `gatsby-remark-images`,
            options: {
              maxWidth: 590,
            },
          },
        ],
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `pages`,
        path: `${__dirname}/src/pages/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `posts`,
        path: `${__dirname}/src/posts/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `assets`,
        path: `${__dirname}/src/assets/`,
      },
    },
    {
      resolve: 'gatsby-plugin-favicon',
      options: {
        logo: './src/images/favicon.png',

        // WebApp Manifest Configuration
        appName: 'Layerise',
        appDescription: null,
        developerName: null,
        developerURL: null,
        dir: 'auto',
        lang: 'en-US',
        background: '#010032',
        theme_color: '#010032',
        display: 'standalone',
        orientation: 'any',
        start_url: '/',
        version: '1.0',

        icons: {
          android: true,
          appleIcon: true,
          appleStartup: true,
          coast: false,
          favicons: true,
          firefox: true,
          opengraph: false,
          twitter: false,
          yandex: false,
          windows: false,
        },
      },
    },
    {
      resolve: `gatsby-plugin-google-analytics`,
      options: {
        trackingId: 'UA-120447155-1',
        // Puts tracking script in the head instead of the body
        head: false,
        // Setting this parameter is optional
        anonymize: true,
        // Setting this parameter is also optional
        respectDNT: true,
        // Avoids sending pageview hits from custom paths
        exclude: ['/preview/**', '/do-not-track/me/too/'],
      },
    },
    {
      resolve: `gatsby-plugin-netlify-cms`,
      options: {
        /**
         * One convention is to place your Netlify CMS customization code in a
         * `src/cms` directory.
         */
        stylesPath: `./src/cms/cms.css`,
      },
    },
  ],
};
```

**Comment by thecodelayeriser** [comment]
Hi, I figured it out :) Will close this.

**Comment by brianrc** [comment]
@thecodelayeriser Can you tell us what you figured out? I think I'm having similar issues.

**Comment by thecodelayeriser** [comment]
Hi Brian, if my memory serves me well I believe I had a typo somewhere. Tried to recall what exactly I miss spelt by can't recall. Sorry for not being able to help more.

**Comment by brianrc** [comment]
Thanks. I think I figured out my issues. Bug in Windows 10 paths caused me so much consternation. Appears to be fixed in gatsby-remark-relative-images 0.2.1, but some starters are not updated yet.
https://github.com/danielmahon/gatsby-remark-relative-images/issues/11
